### DISCUSSION TITLE
캐릭터에 부착된 컴포넌트에 접근이 불가능한 문제를 우회할 방법이 있을까요?

### DISCUSSION THREAD

**Discussion by Chordcode002** [naverz/zepeto-studio-kor#806] [5. Q&A]
this.userIdText.GetComponent<Text>().text = "MyCID : " + ZepetoPlayers.instance.GetPlayer(tempId).character.GetComponent<CharacterBehavior>().id;

모바일 환경에서 제대로 값을 받았는지 확인하기 위해 일일히 텍스트로 표시해가면서 작업하고 있었습니다만
이렇게해서 캐릭터에 부착된 스크립트 컴포넌트에 접근해서 값을 받아와서 표시했습니다.
그런데 PC 에디터 환경에서는 제대로 동작했는데 모바일에서는 NULL인지 표시가 되지 않더라고요.

이전에 비슷한 질문이 있었는데 그 때 답변이 다음과 같았습니다.

'안녕하세요, 런타임에 추가된 커스텀 클래스 컴포넌트일 경우 모바일 테스트시에
GetComponent가 안되는 문제가 보고되어 파악 중에 있습니다.
감사합니다.'

그런데 해당 문제가 해결되기 전까지 기다리고만 있을 수가 없어서요. 혹시 우회라도 가능한 방법이 있을까요?

**Comment by ttoowa** [comment]
저도 GetComponent가 모바일에서만 제대로 되질 않아서 많이 불편하지만
아래와 같은 방법으로 우회해서 사용하고 있습니다

```ts
export default class Src extends ZepetoScriptBehaviour {
    public static InstanceMap: Map<GameObject, Src> = new Map();

    private Awake() {
        Src.InstanceMap.set(this.gameObject, this);
    }
}

export default class SrcManager extends ZepetoScriptBehaviour {
    public GetSrc(gameObject: GameObject): Src {
        return Src.InstanceMap.get(gameObject);
    }
}
```

**Reply by Chordcode002** [reply]
감사합니다 바로 적용해보도록 하겠습니다.

**Reply by IndiaInk10** [reply]
감사합니다

**Comment by ted002** [comment]
저는 그래서 해당 컴포넌트를 그냥 빈 오브젝트를 프리펩으로 만들어서 플레이어가 생성될때 자식으로 주고 있습니다

**Reply by Chordcode002** [reply]
지금 해당 방법을 적용해봤는데, 저는 프리펩으로 미리 만들어둔 경우(런타임에 생성하지 않는다?-라고 판단) 프리펩의 스크립트 오브젝트를 접근할 수 있다라고 판단했습니다만 프리펩의 스크립트에서도 GetComponent가 제대로 동작하지 않는 것 같습니다. 제가 잘못 이해한 걸까요?

아래가 코드입니다

```
//이 코드는 잘 실행됩니다. 자식으로 선언한 프리펩 오브젝트의 이름을 정상적으로 출력합니다.
console.log(ZepetoPlayers.instance.GetPlayer(this.Id).character.GetComponent<CharacterBehavior>().colliders[i].transform.GetChild(1));
//이 코드에서 에러가 납니다.
//TypeError: Cannot read property 'colliders' of null
//이번에도 똑같이 컴포넌트를 못 읽어오는 것 같습니다만....
console.log(ZepetoPlayers.instance.GetPlayer(this.Id).character.GetComponent<CharacterBehavior>().colliders[i].transform.GetChild(1).gameObject.GetComponent<IdHolder>().GetThisChaID());
```

**Reply by Chordcode002** [reply]
생각해보니 여기는 왜 `GetComponent<CharacterBehavior>()`가 되는지 의문이군요 정말 머리 아프네요.
